python/ray/data/block.py

```diff
@@ -166,7 +166,7 @@ def sort_and_partition(self, boundaries: List[T], key: Any,
         """Return a list of sorted partitions of this block."""
         raise NotImplementedError
 
-    def combine(self, key: "GroupKeyT", agg: "AggregateFn") -> Block[U]:
+    def combine(self, key: "GroupKeyT", agg: Tuple["Aggregation"]) -> Block[U]:
```

Comment:
Hmm for `Tuple` --> `Iterable`, we're currently assuming in the `ArrowBlock`/`SimpleBlock` implementations that `aggs` is an indexable sequence rather than an iterable. I'll change it to take a `Sequence`, I think.